### PR TITLE
feat: use __SWR_DEVTOOLS_USE__ hook

### LINKED ISSUE #52: Injecting the middleware automatically
To inject the DevTools middleware to the application, we currently have to wrap it with `<SWRDevTools>` when rendering. This is a great solution but not perfect due to the restrictions:
- Users have to include SWRDevTools in their app bundle to debug production
- The application will always run the SWRDevTools code (even if no SWRDevTools is installed)

It would be great to inject the middleware automatically to all SWR apps, from the extension side. So I opened https://github.com/vercel/swr/issues/2003 to discuss this idea. In extension’s content, we create the middleware and add it to `__SWR_DEVTOOLS_USE__`:

```jsx
global.__SWR_DEVTOOLS_USE__ = global.__SWR_DEVTOOLS_USE__ || []
global.__SWR_DEVTOOLS_USE__.push(devtoolsMiddleware)
```

And then SWR will always use the middleware if defined.

The only thing is to make sure the injection runs before application code to avoid mismatching of number of hooks. I think `"run_at": "document_start"` already does that but not 100% sure.

### PR DESCRIPTION
**Pull request by koba04**
fixes #52 

This requires to add the hook into `swr`, so I'm working on it in https://github.com/koba04/swr/tree/devtools-hook

### CODE REVIEW

**Comment by koba04** [comment]
https://github.com/koba04/swr-devtools/pull/55/commits/f61c092be69f1a5ac704b290bec9aa4745f991dd uses the existing middleware without the use of React Hooks.

**Comment by shuding** [comment]
I'll do a beta release of SWR these two days and we can try it out. ❤️

File: packages/swr-devtools/src/createSWRDevTools.ts [hunk 1]
```diff
@@ -0,0 +1,245 @@
...
+
+export const createSWRDevtools = () => {
+  const events = new EventEmitter();
+
+  const swrdevtools: Middleware = (useSWRNext) => (key, fn, config) => {
+    // use the same React instance with the application
+    const { useLayoutEffect, useEffect, useRef } =
+      typeof window !== "undefined" &&
+      // @ts-expect-error
+      typeof window.__SWR_DEVTOOLS_REACT__ !== "undefined"
```
Comment (huozhi) [inline packages/swr-devtools/src/createSWRDevTools.ts:109]: @koba04 Do you feel we could also use a "hook" or "callback" thing to pass down more stuff? like

```js
const { React: SharedReact } = window.__SWR_DEVTOOLS_HOOK__()
const { useLayoutEffect, useEffect, useRef } = SharedReact
```

once we need anything else besides the shared `React`, we can pass more parameters to the return value of the callback. wdyt?

Comment (shuding) [reply packages/swr-devtools/src/createSWRDevTools.ts:109]: Do you have any example for things other than React that we might need @huozhi?

Comment (huozhi) [reply packages/swr-devtools/src/createSWRDevTools.ts:109]: Not yet, I just feel it's easier to not break the protocol, for now we only need `React`, but later we might have others. Returning from a hook function can be pretty flexible

Comment (koba04) [reply packages/swr-devtools/src/createSWRDevTools.ts:109]: @huozhi @shuding 
As @shuding mentioned, React is a special case for that reason, so we don't have other things we'd like to pass.
But something might come up in the future. In this case, I can add a new "hook" like `__SWR_DEVTOOLS_HOOK__` and include `React` in the hook, so I feel `__SWR_DEVTOOLS_REACT__` is enough at this time.

Comment (huozhi) [reply packages/swr-devtools/src/createSWRDevTools.ts:109]: Yeah that sounds fair. Maybe a renaming to __SWR_DEVTOOLS_XXX__ else not tight to react itself can be nicer. But it's just my personal thought not a blocking at all for any current change 👍 🚀 

If devtool is tighted with version then we can also change it later when it's necessary

**Comment by koba04** [comment]
https://github.com/vercel/swr/pull/2016 has been merged, so I'll merge this too. If there is any issue, I'll fix it later.

**Pull request merged**